src/util/SE3.py

```diff
@@ -1,102 +1,100 @@
 from __future__ import annotations
+from typing import Optional, Union
+from dataclasses import dataclass
 
 import numpy as np
+import tf2_ros
+import rospy
 
-from geometry_msgs.msg import Pose, Transform
 from ros_numpy import numpify
-from tf.transformations import (
-    quaternion_inverse,
-    quaternion_matrix,
-    quaternion_multiply,
-    rotation_from_matrix,
-)
-from .tf_utils import point_to_vector3, vector3_to_point
+from geometry_msgs.msg import TransformStamped, Vector3, Quaternion
+from .SO3 import SO3
 
 
-# TODO: adhere to https://github.com/umrover/mrover-ros/discussions/46
-class SE3(Pose):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+@dataclass
+class SE3:
+    position: np.ndarray
+    rotation: SO3
 
-    @classmethod
-    def from_tf(cls, tf: Transform) -> SE3:
-        """
-        Create an SE3 pose object from a Transform message.
-        The translation of the Transform will become the position of the Pose,
-        and the rotation of the Transform will become the orientation of the Pose.
-
-        :param tf: the Transform message object
-        :returns: the created SE3 pose object
+    def __init__(self, position: np.ndarray = None, rotation: np.ndarray = None):
```

Comment:
dataclass actually automatically creates a constructor for you, you can provide default args: https://docs.python.org/3/library/dataclasses.html